geopandas/geoseries.py

```diff
@@ -30,7 +35,9 @@
 from .base import is_geometry_type
 
 
-def _geoseries_constructor_with_fallback(data=None, index=None, crs=None, **kwargs):
+def _geoseries_constructor_with_fallback(
+    data=None, index=None, crs: Optional[Any] = None, **kwargs
```

Comment:
Just wondering because I am no typing expert, but how useful it is to annotate `crs` with `Optional[Any]` here? (because that basically gives no actual information. `Any` is not the default when untyped?)

Reply:
I can only really speak for mypy, semantically, mypy can track this union type separately:
```python
a : Optional[Any]
reveal_type(a) # note: Revealed type is "Union[Any, None]"
if a:
    reveal_type(a)   # note: Revealed type is "Any"
else:
    reveal_type(a) #  note: Revealed type is "Union[Any, None]"
```
But in practice I don't think this is useful since None is part of the Any type any way. So marking this optional won't help statically, it's only really a reminder to a programmer that it could be none. 

So there are probably two sensible options:
- replace this with purely Any
- Define a `CrsType = Union[int, pyproj.crs.CRS, Any]` which again is effectively just a reminder to a user and hope that in the future at some point pyproj has a more strict annotation on `from_user_input` 


Something else I was wondering which @jorisvandenbossche you might be able to weigh in on given pandas experience (if it has come up at all), should we use `from __future__ import annotations` or not. Originally I thought this would be "safe" since e.g. pandas has been using it for a long while, but I gather PEP 563 is still in limbo

Reply:
Here's a [discussion](https://github.com/python/mypy/issues/3138) on this from my-yes repo, concluding that `Optional[Any]` is not the same as `Any`. I am personally fine keeping `Optional[Any]` in here, rather than specifying a subset like `Union[int, pyproj.crs.CRS, Any]` as it would have to be more something like `Union[int, str, dict, tuple pyproj.crs.CRS, Any]` at which point, `[Any]` is probably better.

> should we use from __future__ import annotations or not

afaik it is generally being used across the ecosystem (e.g. xarray).